Source/ablastr/mesh_refinement/coarsenToCellCentered.H

```diff
@@ -1,26 +1,36 @@
-#ifndef WARPX_COARSEN_IO_H_
-#define WARPX_COARSEN_IO_H_
+/* Copyright 2022 Edoardo Zoni, Remi Lehe, David Grote, Axel Huebl
+ *
+ * This file is part of ABLASTR.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+#ifndef ABLASTR_MR_COARSEN_TO_CELL_CENTERED_H_
+#define ABLASTR_MR_COARSEN_TO_CELL_CENTERED_H_
+
 
 #include <AMReX_Array.H>
 #include <AMReX_Array4.H>
+#include <AMReX_BLassert.H>
 #include <AMReX_Extension.H>
 #include <AMReX_GpuQualifiers.H>
 #include <AMReX_IntVect.H>
+#include <AMReX_Math.H>
 #include <AMReX_REAL.H>
 
 #include <AMReX_BaseFwd.H>
 
 #include <cstdlib>
 
-namespace CoarsenIO{
-
-    using namespace amrex;
 
+namespace ablastr::mesh_refinement::to_cell_centered
+{
     /**
      * \brief Interpolates the floating point data contained in the source Array4
      *        \c arr_src, extracted from a fine MultiFab, by averaging over either
      *        1 point or 2 equally distant points.
      *
+     * The output (sc) staggering is always cell-centered.
```

Comment:
@EZoni I think I might be wrong with this detail. Let us clarify so its clear what the difference is to the other function.

Reply:
In this case I don't think the output index type has to be cell-centered. We do use only cell-centered data for I/O, but I think this `Interp` function was written in a general way, such that it works for an arbitrary output index type (nodal or cell-centered). 

Thus, I would suggest two changes (can be discussed further, if necessary):
- remove the `AMREX_ASSERT`s added below, as I think the function is expected to return a correct result even if the output index type is nodal;
- rename the file differently, since this is not specialized for cell-centered output only.

Alternatively, if we want this to be specialized for cell-centered output only, we could also clean up the formulas by setting `sc` to zero everywhere. But I'd be more in favor of the solution above.

Reply:
Thanks for checking and the details, @EZoni!

I think all that is left is finding a good name for this - I think coarsen "MR"/"IO" is not very descriptive right now...